cli/src/semdep/parsers/yarn.py

```diff
@@ -0,0 +1,245 @@
+"""
+Parsers for yarn.lock versions 1 and 2/3
+"""
+from pathlib import Path
+from typing import List
+from typing import Optional
+from typing import Set
+from typing import Tuple
+from typing import TypeVar
+
+from parsy import any_char
+from parsy import Parser
+from parsy import peek
+from parsy import string
+from parsy import success
+
+from semdep.parsers.util import consume_line
+from semdep.parsers.util import extract_npm_lockfile_hash
+from semdep.parsers.util import json_doc
+from semdep.parsers.util import mark_line
+from semdep.parsers.util import pair
+from semdep.parsers.util import quoted
+from semdep.parsers.util import safe_path_parse
+from semdep.parsers.util import transitivity
+from semdep.parsers.util import upto
+from semgrep.semgrep_interfaces.semgrep_output_v1 import Ecosystem
+from semgrep.semgrep_interfaces.semgrep_output_v1 import FoundDependency
+from semgrep.semgrep_interfaces.semgrep_output_v1 import Npm
+from semgrep.verbose_logging import getLogger
+
+logger = getLogger(__name__)
+
+A = TypeVar("A")
+
+# The initial line of a yarn version 1 dependency, lists the constraints that lead to this package
+# Examples:
+# "@ampproject/remapping@^2.0.0"
+# bad-lib@0.0.8
+def source1(quoted: bool) -> "Parser[Tuple[str,str]]":
+    return (
+        string("@")
+        .optional(default="")
+        .bind(
+            lambda at_prefix: upto("@", consume_other=True).bind(
+                lambda package: upto(
+                    *(['"', ","] + ([":"] if not quoted else []))
```

Comment:
Can you add examples of when there are : and , in the line

Reply:
Done